src/EditorFeatures/Core/GoToNextAndPreviousMethodCommandHandler.cs

```diff
@@ -0,0 +1,148 @@
+            => ExecuteCommand(nextHandler, args.SubjectBuffer, args.TextView, next: false);
+
+        private static CommandState GetCommandState(ITextBuffer subjectBuffer, Func<CommandState> nextHandler)
+        {
+            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null || !document.SupportsSyntaxTree)
+            {
+                return nextHandler();
+            }
+
+            return CommandState.Available;
+        }
+
+
+        private void ExecuteCommand(Action nextHandler, ITextBuffer subjectBuffer, ITextView textView, bool next)
+        {
+            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null || !document.SupportsSyntaxTree)
+            {
+                nextHandler();
+                return;
+            }
+
+            var caretPoint = textView.GetCaretPoint(subjectBuffer);
+            if (!caretPoint.HasValue)
+            {
+                nextHandler();
+                return;
+            }
+
+            int? targetPosition = null;
+            var waitResult = _waitIndicator.Wait("", allowCancel: true, action: waitContext =>
+            {
+                targetPosition = GetTargetPosition(document, caretPoint.Value.Position, next, waitContext.CancellationToken);
+            });
+
+            if (waitResult == WaitIndicatorResult.Canceled || targetPosition == null)
+            {
+                return;
+            }
+
+            // TODO: Better position within the node (e.g. attributes?)
+            var targetLocation = new SnapshotPoint(subjectBuffer.CurrentSnapshot, targetPosition.Value);
+            var viewLocation = textView.BufferGraph.MapUpToBuffer(targetLocation, PointTrackingMode.Positive, PositionAffinity.Successor, textView.TextBuffer);
+            if (viewLocation.HasValue)
+            {
+                textView.Caret.MoveTo(viewLocation.Value);
+            }
+        }
+
+        private static int? GetTargetPosition(Document document, int caretPosition, bool next, CancellationToken cancellationToken)
+        {
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+            if (syntaxFactsService == null)
+            {
+                return null;
+            }
+
+            var root = document.GetSyntaxRootAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var currentMember = GetMember(syntaxFactsService, caretPosition, root);
+            if (currentMember == null)
+            {
+                return null;
+            }
+
+            var members = syntaxFactsService.GetMethodLevelMembers(root);
+            var index = members.IndexOf(currentMember);
+            if (index < 0)
+            {
+                Debug.Fail("Couldn't find current member in members?");
```

Comment:
Well, I would expect `currentMember` to have been null in that case.  However, I do need to test what happens when you're between members.